homeassistant/requirements.py

```diff
@@ -97,12 +97,18 @@ async def async_get_integration_with_requirements(
             deps_to_check.append(check_domain)
 
     if deps_to_check:
-        await asyncio.gather(
-            *[
-                async_get_integration_with_requirements(hass, dep, done)
-                for dep in deps_to_check
-            ]
-        )
+        try:
+            await asyncio.gather(
```

Comment:
Should this be using `return_exceptions=True`?

Reply:
You may be right but I think it's a corner case since not many integrations have multiple dependencies/after_dependencies. I guess the issue would be an after_dependency returns first and thus the returned exception isn't treated as a integrationnotfound? The other direction is fine because dependencies are required and should fail on the first.

The [documentation](https://docs.python.org/3/library/asyncio-task.html#running-tasks-concurrently) confuses me because it says the other awaitables continue. Doesn't that mean they can also throw an exception which can be caught later?

> If return_exceptions is False (default), the first raised exception is immediately propagated to the task that awaits on gather(). Other awaitables in the aws sequence won’t be cancelled and will continue to run.

Reply:
The other ones will continue to run and if they throw an exception you'll probably not see it since nothing will every await the task and raise it.

Reply:
Ok, change made.